acceptance-tests/src/test/scala/ch/renku/acceptancetests/pages/ProjectPage.scala

```diff
@@ -361,51 +359,24 @@ class ProjectPage(val projectSlug: String, val namespace: String)
           .getOrElse(fail("Sessions -> Running: title not found"))
       }
 
-      def connectToJupyterLab(implicit webDriver: WebDriver, spec: AcceptanceSpec): JupyterLabPage =
-        connectToJupyterLab(s"a.dropdown-item[href*='/sessions/']", s"div.sessionsButton > button")
-
-      def connectToAnonymousJupyterLab(implicit webDriver: WebDriver, spec: AcceptanceSpec): JupyterLabPage =
-        connectToJupyterLab(s"a.dropdown-item[href*='/sessions/']", s"div.sessionsButton > button")
+      def maybeOpenButton(implicit webDriver: WebDriver): Option[WebElement] = eventually {
+        find(cssSelector("a[href*='sessions/show']"))
+          .find(_.text.trim == "Open")
+      }
 
-      def connectButton(buttonSelector: String)(implicit webDriver: WebDriver): WebElement = eventually {
-        find(
-          cssSelector(buttonSelector)
-        ) getOrElse fail(
-          "First row session Connect button not found"
-        )
+      def maybeGetLogsButton(implicit webDriver: WebDriver): Option[WebElement] = eventually {
```

Comment:
Is this "maybeGetLogsButton" visible to the test writer? I'm not sure if people coming from dynamic languages would necessarily understand that this means Option

Reply:
Yes, maybe you are right. I stopped writing non-functional code some time ago and it's somehow natural to me to use the naming and structures. On the other side, I don't know what could be a better and still meaningful name. And maybe it's also a good thing to give our colleagues a bit of a touch of FP ;)